modules/custom/openy_activity_finder/openy_cf_vue_app/components/CampFinder.vue

```diff
@@ -0,0 +1,514 @@
+        for (var topLevelCategories in this.activities) {
+          for (var category in this.activities[topLevelCategories].value) {
+            if (this.activities[topLevelCategories].value[category].value == value) {
+              return this.activities[topLevelCategories].value[category].label;
+            }
+          }
+        }
+      },
+      getCategoryParentNameByChildId: function(value) {
+        for (var topLevelCategories in this.activities) {
+          for (var category in this.activities[topLevelCategories].value) {
+            if (this.activities[topLevelCategories].value[category].value == value) {
+              return this.activities[topLevelCategories].label;
+            }
+          }
+        }
+      },
+      getLocationNameById: function(value) {
+        for (var topLevelLocation in this.locations) {
+          for (var location in this.locations[topLevelLocation].value) {
+            if (this.locations[topLevelLocation].value[location].value == value) {
+              return this.locations[topLevelLocation].value[location].label;
+            }
+          }
+        }
+      },
+      getAgeNameById: function(value) {
+        for (var i in this.ages) {
+          if (this.ages[i].value == value) {
+            return this.ages[i].label;
+          }
+        }
+      },
+      getWeekNameById: function(value) {
+        for (var i in this.weeks) {
+          if (this.weeks[i].value == value) {
+            return this.weeks[i].label;
+          }
+        }
+      },
+      initializeFromGet: function() {
+        if (typeof this.$route.query.ages != 'undefined') {
+          var checkedAgesGet = decodeURIComponent(this.$route.query.ages);
+          if (checkedAgesGet !== '') {
+            this.checkedAges = checkedAgesGet.split(',');
+          }
+        }
+
+        if (typeof this.$route.query.weeks != 'undefined') {
+          var checkedWeeksGet = decodeURIComponent(this.$route.query.weeks);
+          if (checkedWeeksGet !== '') {
+            this.checkedWeeks = checkedWeeksGet.split(',');
+          }
+        }
+
+        if (typeof this.$route.query.categories != 'undefined') {
+          var checkedCategoriesGet = decodeURIComponent(this.$route.query.categories);
+          if (checkedCategoriesGet !== '') {
+            this.checkedCategories = checkedCategoriesGet.split(',');
+          }
+        }
+
+        if (typeof this.$route.query.locations != 'undefined') {
+          var checkedLocationsGet = decodeURIComponent(this.$route.query.locations);
+          if (checkedLocationsGet !== '') {
+            this.checkedLocations = checkedLocationsGet.split(',');
+          }
+        }
+
+        if (typeof this.$route.query.initial != 'undefined') {
+          let initialStepGet = decodeURIComponent(this.$route.query.initial);
+          if (initialStepGet !== '') {
+            this.initialStep = initialStepGet;
+          }
+        }
+      },
+      getHumanReadableAppliedFilters() {
```

Comment:
Why isn't this method in "computed" section? It seems like it depends only on component state and should be there in computed properties.

Reply:
I prefer to keep it in methods section for now